src/notify.js

```diff
@@ -71,6 +83,11 @@ Notify.isSupported = function(perm) {
         throw new Error('You must only call this before calling Notification.requestPermission(), otherwise this feature detect would trigger an actual notification!');
     }
 
+    if (Notify.serviceWorkerSupported()) {
```

Comment:
OK, how about this then:

```javascript
Notify.isSupported = function(usingServiceWorker = Notify.isServiceWorker()) {
    ...
};
```

I agree with you now though, the problem is more complicated than I thought